files/restore_backup.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python
+
+import os
+import re
+import shutil
+
+DOCUMENTATION = '''
+---
+module: restore_backup
+short_description: restores a backup created with 'backup=true' using, for instance, the lineinfile module
+description:
+  - looks for a backup file in the same dir as the file to be restored and renames it to the original.
+author: Yoram Hekma
+version_added: 2.0
+requirements:
+  - python > 2.6
+options:
+  dest:
+    description:
+      - The name of the file that needs to be restored.
+    required: true
+  backup:
+    description:
+      - Create a backup file including the timestamp information so you can get
+        the original file back if you somehow clobbered it incorrectly.
+    required: false
+    choices: ["yes", "no"]
+    default: "no"
+'''
+
+EXAMPLES = '''
+  - action: restore_backup dest=/etc/samba/smb.conf
+'''
+
+
+def restore_backup(m, dest):
```

Comment:
I think I would recommend, still calling this `module` instead of `m`.  Short names like that make code more difficult to understand.  

This is a common approach to pass `module` in so you have access to it, so +1 there.